### DISCUSSION TITLE
Future features (maybe)

### DISCUSSION THREAD

**Discussion by love98ooo** [NJUPT-SAST/sast-evento-backend#79] [Ideas]
We want to give EVENTO more possibilities, and that means we will add more practical or creative functions for it.

Such as:

1. searching event based on titles and descriptions
2. adding covers for all events
3. adding customized avatar for both all groups and users
4. ......

---

You can give your suggestions in this discussion or open a new discussion with `Ideas` tag.

**Comment by maplestarplayl** [comment]
关于第一点，如果仅仅只是根据标题和叙述来搜索，那MySQL自带的[full-text-search](https://dev.mysql.com/doc/refman/8.2/en/fulltext-search.html)就行,实现起来也不难，建库的时候加条FULLTEXT语句就行，然后再加个用户的接口应该就行了
[至于为什么不直接用LIKE](https://stackoverflow.com/questions/224714/what-is-full-text-search-vs-like)，现阶段用也不是不行，但是以后不方便优化，而full-text-search有很多微调的空间，也就是用精准度去换取更多的匹配结果。而且full-text-search也能建立索引，速度上会更快
抛砖引玉下，有没有人来讨论下😘

**Reply by love98ooo** [reply]
#84 这个问题可以到这里讨论

**Comment by feellmoose** [comment]
优化建议:
多表查询分离+缓存
提升查询速度，缓解mysql压力

**Comment by feellmoose** [comment]
建议课程相关事件可以增加：
1. 课件附件链接
2. 课程直播或回放链接

**Comment by feellmoose** [comment]
考虑到未来和event相关的组件逐渐增多，是否需要使用发布订阅机制（事件或消息）进行组件间解耦

相关设计：
1. DDD：领域事件
2. 消息驱动
3. 事件驱动

具体过程：
1. 发布“发布event事件”，发布事件后即完成发布
2. 各组件接收“event事件”，并执行各自逻辑

**Reply by feellmoose** [reply]
目前常用的事件组件：1. spring-event 2. vertx-eventbus 3. 外部的消息或事件队列
